pkg/model/resource/utils.go

```diff
@@ -67,7 +65,29 @@ func APIPackagePathLegacy(repo, group, version string, multiGroup bool) string {
 	return path.Join(repo, "api", version)
 }
 
+// unpluralizedSuffixes is a list of resource suffixes that are the same plural and singular
+// This is only is only necessary because some bits of code are lazy and don't actually use the RESTMapper like they should.
+// TODO eliminate this so that different callers can correctly map to resources.  This probably means updating all
+// callers to use the RESTMapper they mean.
+var unpluralizedSuffixes = []string{
+	"endpoints",
+}
+
 // RegularPlural returns a default plural form when none was specified
+// Reference: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/restmapper.go#L126
 func RegularPlural(singular string) string {
-	return flect.Pluralize(strings.ToLower(singular))
+	singularName := strings.ToLower(singular)
+	for _, skip := range unpluralizedSuffixes {
+		if strings.HasSuffix(singularName, skip) {
+			return singular
+		}
+	}
+
+	switch string(singularName[len(singularName)-1]) {
+	case "s":
+		return singularName + "es"
+	case "y":
+		return strings.TrimSuffix(singularName, "y") + "ies"
```

Comment:
There is a piece of code in k8s/code-generator, that deals with this and some of the edge cases explicitly: https://github.com/kubernetes/gengo/blob/ab3349d207d4/namer/plural_namer.go#L36. May not be an immediate action item, but it would be preferable to have a TODO added here, to be able to resuse that piece of code. (It currently is very opinionated to spit out plurals for core k8s resources).